pynitrokey/cli/start.py

```diff
@@ -198,11 +221,28 @@ def update(
             "use one from: https://github.com/Nitrokey/nitrokey-start-firmware)",
         )
 
-    if IS_LINUX:
-        with ThreadLog(logger.getChild("dmesg"), "dmesg -w"):
+    with confirm_keyboard_interrupt("Cancelling the update may brick your device."):
+        if IS_LINUX:
+            with ThreadLog(logger.getChild("dmesg"), "dmesg -w"):
+                start_update(*args)
+        else:
             start_update(*args)
-    else:
-        start_update(*args)
+
+
+def find_udev_rules() -> bool:
+    dirs = [
+        "/usr/lib/udev/rules.d",
+        "/usr/local/lib/udev/rules.d",
+        "/run/udev/rules.d",
+        "/etc/udev/rules.d",
+    ]
+    for d in dirs:
+        if os.path.isdir(d):
+            for name in os.listdir(d):
+                if fnmatch.fnmatch(name, "??-nitrokey.rules"):
```

Comment:
AFAIR using anything starting with number over `70-` will probably not work. At some point we had a rule starting with `99-`, which brought a couple of issues.
I would change this to accept only the official name, which starts with `41-`, to avoid potential problems.
Ideally it would be really nice to check for the updates of the udev file and compare hashes to determine, whether it is up to date.